nixos/modules/services/databases/mysql.nix

```diff
@@ -307,9 +299,14 @@ in
           pkgs.nettools
         ];
 
-        preStart = ''
+        preStart = if isMariaDB then ''
+          if ! test -e ${cfg.dataDir}/mysql; then
+            ${mysql}/bin/mysql_install_db --defaults-file=/etc/my.cnf ${mysqldOptions}
+            touch /tmp/mysql_init
+          fi
+        '' else ''
           if ! test -e ${cfg.dataDir}/mysql; then
-            ${mysql}/bin/mysql_install_db --defaults-file=/etc/my.cnf ${installOptions}
+            ${mysql}/bin/mysqld --defaults-file=/etc/my.cnf ${mysqldOptions} --initialize-insecure
             touch /tmp/mysql_init
           fi
```

Comment:
this is slightly duplicated, why not do
```
preStart = 
let
  initCommand = if isMariaDB then ''
    ${mysql}/bin/mysql_install_db --defaults-file=/etc/my.cnf ${mysqldOptions}
  '' else "${mysql}/bin/mysqld --defaults-file=/etc/my.cnf ${mysqldOptions} --initialize-insecure";
in ''
  if ! test -e ${cfg.dataDir/mysql; then
    ${initCommand}
    touch /tmp/mysql_init
  fi
'';
```

Reply:
As time goes by `mariadb` and `mysql` drift further apart so I figured in this case maybe just keep it separate in case there is more divergence... but yeah maybe it was a bit much and a bit premature.

Reply:
> As time goes by mariadb and mysql drift further apart so I figured in this case maybe just keep it separate in case there is more divergence...

That is a good point, this is probably fine if the actual conditional changes.